alibi_detect/base.py

```diff
@@ -229,6 +225,17 @@ def from_config(self): ...
     def _set_config(self): ...
 
 
+@runtime_checkable
+class StatefulDetector(Detector, Protocol):
```

Comment:
😎 Proto-cool!

Reply:
We should probably rename these `DetectorProtocol`, `ConfigurableDetectorProtocol` and `StatefulDetectorProtocol`. Otherwise we risk people thinking `Detector` is the base detector object and for some reason these protocols inherit from it!

Reply:
Also should `StatefulDetector` inherit from `ConfigureableDetector`? Are all stateful detectors configurable? If not are they intended to be at some point?

Reply:
Ha ha you have converted me into a  `Protocol` fan!

Good point about the renaming and inheritance. I'll do the inheritance once now, and have added an issue to rename the `Protocol`'s since renaming `Detector` and `ConfigurableDetector` isn't really related to online state.

Reply:
Issue: https://github.com/SeldonIO/alibi-detect/issues/707